src/auth.js

```diff
@@ -2,25 +2,27 @@ import { signInWithEmailAndPassword, createUserWithEmailAndPassword, signOut } f
 import { auth } from './firebase'
 
 // USER LOGIN
-export const login = async (credentials, navigate) => {
+export const login = async (credentials, navigate, errorMessage, popUp) => {
   const loginData = { ...credentials }
   try {
     const loginUser = await signInWithEmailAndPassword(auth, loginData.email, loginData.password)
     if (loginUser) navigate('/chat')
+    popUp()
   } catch (error) {
     console.log(error.message)
-    alert('Email or password is wrong, please try again')
+    errorMessage()
   }
 }
 // REGISTER USER
-export const register = async (credentials, navigate) => {
+export const register = async (credentials, navigate, errorMessage, popUp) => {
   const registerData = { ...credentials }
   try {
     const registerUser = await createUserWithEmailAndPassword(auth, registerData.email, registerData.password)
     if (registerUser) navigate('/chat')
+    popUp()
   } catch (error) {
     console.log(error.message)
-    alert('Email or password are not correctly written, please try again')
+    errorMessage()
```

Comment:
After configuring the toast container, you can just use:
`toast.error('Email or password is wrong, please try again')`